src/coreclr/pal/src/exception/machexception.cpp

```diff
@@ -1416,192 +1416,4 @@ SEHInitializeMachExceptions(DWORD flags)
     return TRUE;
 }
 
-extern "C"
-void
-ActivationHandler(CONTEXT* context)
-{
-    if (g_activationFunction != NULL)
-    {
-        g_activationFunction(context);
-    }
-
-#ifdef TARGET_ARM64
-    // RtlRestoreContext assembly corrupts X16 & X17, so it cannot be
-    // used for Activation restore
-    MachSetThreadContext(context);
```

Comment:
How is context restored for activation using signals?  How are we preventing X16/X17 corruption?

Reply:
I am actually wondering if we should change the JIT to not use X17 unless it is marked unsafe for GC.

Reply:
The context is restored by the kernel. We return from the signal handler and let the kernel do the job. We just copy CONTEXT to the ucontext_t passed to the signal handler before returning.